src/main/scala/org/constellation/primitives/ConcurrentTipService.scala

```diff
@@ -213,4 +215,43 @@ class ConcurrentTipService[F[_]: Concurrent: Logger: Clock](
       }
     selectedFacils.toMap
   }
+
+  private def filterFacilitators(facilitators: Map[Id, PeerData]) =
+    for {
+      facilitatorHeights <- LiftIO[F].liftIO(getFacilitatorHeights(facilitators))
+      ownHeight <- ownHeight()
+      _ <- Logger[F].debug(s"filter facilitators : own height = $ownHeight : facilitatorHeights = $facilitatorHeights")
+
+      facilitatorsIds <- filterByHeight(facilitatorHeights, maxOrZero(ownHeight))
+      _ <- Logger[F].debug(s"facilitators ids = $facilitatorsIds")
+
+      filteredFacilitators <- Sync[F].delay { facilitators.filter(f => facilitatorsIds.contains(f._1)) }
+      _ <- Logger[F].info(s"filtered facilitators = $filteredFacilitators")
+    } yield filteredFacilitators
+
+  private def getFacilitatorHeights(facilitators: Map[Id, PeerData]) =
+    facilitators.toList
+      .traverse(
+        p =>
+          (
+            p._1,
+            p._2.client
+              .getNonBlockingIO[List[Height]]("heights")(IO.contextShift(ConstellationExecutionContext.bounded))
```

Comment:
shouldn't the context be passed from consensus? it's hardcoded there but in consensus we're using unbounded probably, which means, part of the consensus will be on unbounded thread, then it will go back to bounded after the api call